generated/src/aws-cpp-sdk-s3/source/S3Client.cpp

```diff
@@ -2027,9 +2027,7 @@ Aws::String S3Client::GeneratePresignedUrl(const Aws::String& bucket,
         return {};
     }
     Aws::Endpoint::AWSEndpoint& endpoint = computeEndpointOutcome.GetResult();
-    URI uri(endpoint.GetURL());
-    uri.SetPath(uri.GetPath() + "/" + key);
-    endpoint.SetURL(uri.GetURIString());
+    endpoint.SetURL(endpoint.GetURL() + "/" + key);
```

Comment:
Hey Thanks for taking the time to contribute! Was able to replicate and take a look at this solution. just a few things about this.

* thinking we should go with `endpoint.AddPathSegments(key);` instead of this as to avoid creating the temporary URI object all together. Also this is how we handle this in [`GetObject` correctly](https://github.com/aws/aws-sdk-cpp/blob/main/generated/src/aws-cpp-sdk-s3/source/S3Client.cpp#L1028) so we should follow the same pattern. The reason the problem arose intially is because these diverged, and this results in a accidental double encoding when calling `uri.GetURIString()`.

i.e.

```
Aws::Endpoint::AWSEndpoint& endpoint = computeEndpointOutcome.GetResult();
endpoint.AddPathSegments(key);
return AWSClient::GeneratePresignedUrl(endpoint, method, customizedHeaders, expirationInSeconds);
```

* you will also need to update the [code-gen file that has this line in it](https://github.com/aws/aws-sdk-cpp/blob/main/tools/code-generation/generator/src/main/resources/com/amazonaws/util/awsclientgenerator/velocity/cpp/s3/S3ClientSource.vm#L124-L126). We run a [codegenerator](https://github.com/aws/aws-sdk-cpp/blob/main/docs/CMake_Parameters.md#regenerate_clients) to generate aws clients based on modeled json. Since S3 is so specialized though, we have singular location for it.

Reply:
Hey, already made the new commit following your Suggestion.

Reply:
nice let me run some tests and make sure its all good, then we should be able to merge